components/Card/Card.test.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import Card from "./Card";
+
+const mockMedication = {
+  id: 1,
+  name: "Medication 1",
+  summary: "Summary of Medication 1",
+  mechanismOfAction: "Mechanism of Action",
+  uses: "Uses of Medication",
+  sideEffects: "Side Effects of Medication",
+  dosageAndAdministration: "Dosage and Administration",
+  precautions: "Precautions for Medication",
+};
+
+describe("Card", () => {
+  it("renders the card with medication details", () => {
+    render(
+      <Card
+        medication={mockMedication}
+        toggleBookmark={() => {}}
+        isDarkMode={false}
+      />
+    );
+
+    // Assert that the medication name is rendered
+    const medicationName = screen.getByText("Medication 1");
+    expect(medicationName).toBeInTheDocument();
+
+    // Assert that the medication summary is rendered
+    const medicationSummary = screen.getByText("Summary of Medication 1");
+    expect(medicationSummary).toBeInTheDocument();
+
+    // Assert that the "More Info" button is rendered
+    const moreInfoButton = screen.getByText("More Info ⌵");
+    expect(moreInfoButton).toBeInTheDocument();
+
+    // Assert that the expanded details are not initially rendered
+    const mechanismOfAction = screen.queryByText("Mechanism of Action");
+    expect(mechanismOfAction).not.toBeInTheDocument();
+  });
+
+  it('toggles the expanded details when the "More Info" button is clicked', () => {
+    render(
+      <Card
+        medication={mockMedication}
+        toggleBookmark={() => {}}
+        isDarkMode={false}
+      />
+    );
+
+    // Click the "More Info" button
+    const moreInfoButton = screen.getByText("More Info ⌵");
+    fireEvent.click(moreInfoButton);
+
+    // Assert that the expanded details are rendered
+    const mechanismOfAction = screen.getByText("Mechanism of Action");
+    expect(mechanismOfAction).toBeInTheDocument();
+
+    // Assert that the "Less Info" button is displayed after clicking
+    const lessInfoButton = screen.getByText("Less Info ⌃");
+    expect(lessInfoButton).toBeInTheDocument();
+
+    // Click the "Less Info" button
+    fireEvent.click(lessInfoButton);
+
+    // Assert that the expanded details are not rendered
+    const mechanismOfActionAfterCollapse = screen.queryByText(
+      "Mechanism of Action"
+    );
+    expect(mechanismOfActionAfterCollapse).not.toBeInTheDocument();
+  });
+});
```

Comment:
```jsx
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Card from "./Card";

const mockMedication = {
  id: 1,
  name: "Medication 1",
  summary: "Summary of Medication 1",
  mechanismOfAction: "Mechanism of Action",
  uses: "Uses of Medication",
  sideEffects: "Side Effects of Medication",
  dosageAndAdministration: "Dosage and Administration",
  precautions: "Precautions for Medication",
};

test("renders the card with medication details", () => {
  render(<Card medication={mockMedication} />);

  const medicationName = screen.getByText("Medication 1");
  expect(medicationName).toBeInTheDocument();

  const medicationSummary = screen.getByText("Summary of Medication 1");
  expect(medicationSummary).toBeInTheDocument();

  const moreInfoButton = screen.getByText("More Info ⌵");
  expect(moreInfoButton).toBeInTheDocument();
});

test("toggles the expanded details when the "More Info" button is clicked", async () => {
  const user = userEvent.setup();

  render(<Card medication={mockMedication} />);

  const moreInfoButton = screen.getByRole("button", { name: "More Info ⌵" });
  await user.click(moreInfoButton);

  const mechanismOfAction = screen.getByText("Mechanism of Action");
  expect(mechanismOfAction).toBeInTheDocument();

  const usesOfMedication = screen.getByText("Uses of Medication");
  expect(usesOfMedication).toBeInTheDocument();

  const sideEffects = screen.getByText("Side Effects of Medication");
  expect(sideEffects).toBeInTheDocument();

  const dosageAndAdministration = screen.getByText("Dosage and Administration");
  expect(dosageAndAdministration).toBeInTheDocument();

  const precautions = screen.getByText("Precautions for Medication");
  expect(precautions).toBeInTheDocument();

  const lessInfoButton = screen.getByText("Less Info ⌃");
  expect(lessInfoButton).toBeInTheDocument();

  await user.click(lessInfoButton);

  expect(mechanismOfAction).not.toBeInTheDocument();
  expect(usesOfMedication).not.toBeInTheDocument();
  expect(sideEffects).not.toBeInTheDocument();
  expect(dosageAndAdministration).not.toBeInTheDocument();
  expect(precautions).not.toBeInTheDocument();
});

```

I refactored also this test according to how we teached you to write tests :) Some further notes:
- your test medication data should be more realistic: "Medication1", "Summary of Medication", etc. are not proper example values.
- as your second test tests which information are sdisplayed, you do not need to test in your first test what informations is not displayed

Reply:
Thank you for your feedback. I appreciate your suggestion, I have already apply all your suggetions